### PR TITLE
Feat: Datetime validator precision

### PR DESCRIPTION
**Pull request by Meldiron**

### CODE REVIEW

File: src/Database/Validator/Datetime.php [hunk 1]
```diff
@@ -29,7 +44,23 @@ public function __construct(?bool $requireDateInFuture = null)
      */
     public function getDescription(): string
     {
-        return $this->message;
+        $message = 'Value must be valid date';
+
+        if($this->offset < 0) {
+            if($this)
```
Comment (christyjacob4) [inline src/Database/Validator/Datetime.php:50]: `if this` ??

File: src/Database/Validator/Datetime.php [hunk 2]
```diff
@@ -6,21 +6,36 @@
...
+
+    /**
+     * @var int minimum offset from now in seconds
+     */
+    protected int $offset = 0;
+
+    /**
+     * @param bool $offset minimum offset from now in seconds
+     */
+    public function __construct(bool $requireDateInFuture = false, string $precision = self::PRECISION_ANY, int $offset = 0)
```
Comment (christyjacob4) [inline src/Database/Validator/Datetime.php:34]: precision and offset are only relevant if requireDateInFuture = true. 
So they can be both optional params

File: src/Database/Validator/Datetime.php [hunk 3]
```diff
@@ -49,11 +80,45 @@ public function isValid($value): bool
...
+                $diff = $date->getTimestamp() - $now->getTimestamp();
+                if($diff <= $this->offset) {
+                    return false;
+                }
+            } else if($this->offset < 0) {
+                $diff = $now->getTimestamp() - $date->getTimestamp();
+                if($diff <= \abs($this->offset)) {
+                    return false;
+                }
+            }
```
Comment (christyjacob4) [inline src/Database/Validator/Datetime.php:96]: offset should be +ve only so we can limit the amount of edge cases.

File: src/Database/Validator/Datetime.php (hunk 1 above)
Comment (Meldiron) [reply src/Database/Validator/Datetime.php:50]: copy&paste mistake, thanks

File: src/Database/Validator/Datetime.php (hunk 2 above)
Comment (Meldiron) [reply src/Database/Validator/Datetime.php:34]: They have default value, and that makes both optional params

File: src/Database/Validator/Datetime.php (hunk 3 above)
Comment (Meldiron) [reply src/Database/Validator/Datetime.php:96]: Removed support for negative offset

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**